### ISSUE TITLE
Slot props cause no-undef errors

### ISSUE THREAD

**Issue by cyderize** [sveltejs/eslint-plugin-svelte3#30]
For some reason I'm getting no-undef errors when using slot props:

Referencing the slot prop `slotProp` in `Foo.svelte` leads to the eslint error `'slotProp' is not defined`. I'm able to work around this by adding `/* global slotProp */`.

**Foo.svelte**
```jsx
<script>
    import Bar from './Bar.svelte';
</script>

<Bar let:slotProp>
    {slotProp}
</Bar>
```

**Bar.svelte**
```jsx
<slot slotProp="bar"/>
```

Could this have to do with my configuration? These only started showing up since I upgraded to v2.4.1 (I was on v1.2.3 before and I didn't get these)

**Comment by Conduitry** [comment]
Thanks for the report! Yeah it looks like the `let:` directive isn't being considered when deciding what's in scope.

**Comment by Conduitry** [comment]
This should be fixed in 2.4.2.

**Comment by cyderize** [comment]
Thanks for the fast fix! Although I think this problem still occurs for named slots (I'm assuming because the let directive is then on a normal element and not an InlineComponent):

**Foo.svelte**
```jsx
<script>
    import Bar from './Bar.svelte';
</script>

<Bar>
  <div slot="mySlot" let:slotProp>
    {slotProp}
  </div>
</Bar>
```

**Bar.svelte**
```jsx
<slot name="mySlot" slotProp="bar"/>
```

**Comment by Conduitry** [comment]
Ah yep, I didn't take care of that case. I will look at that tomorrow. Thanks!

**Comment by Conduitry** [comment]
That second case with `slot=` should be fixed in 2.5.0.

**Comment by cyderize** [comment]
Yep, it works great now, thanks!
